src/Cinelovers.Core/Caching/ICache.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Reactive;
+
+namespace Cinelovers.Core.Caching
+{
+    public interface ICache
+    {
+        void Initialize(string name);
+        IObservable<TResult> GetAndFetchLatest<TResult>(string cacheKey, Func<IObservable<TResult>> fetchFunction);
+        IObservable<Unit> InvalidateAll();
```

Comment:
Any special reason to map all these methods that are never used?

Reply:
Not really, actually I could clean-up these but I decided to maintain them because they are commonly used. Anyway, I can remove them and re-expose them as needed.